### DISCUSSION TITLE
[ZK Stack] - Transactions stuck at Validtion step

### DISCUSSION THREAD

**Discussion by sachanayush47** [zkSync-Community-Hub/zksync-developers#727] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [X] Server (zksync-era)
- [X] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [X] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.80.1

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

Using the tutorial https://code.zksync.io/tutorials/custom-zk-chain/customizing-your-chain
I have deployed a Validium based Zk hyperchain with custom base token, holesky as L1 and mock gpu prover.
Problem Im facing is transactions are stuck at validation step. Additionally, L2 to L1 are not working.

We using Infura Paid RPC.

![image](https://github.com/user-attachments/assets/7de27074-5f65-4cd3-bf18-abca4d323c1e)

![image](https://github.com/user-attachments/assets/d47b9ac2-f34d-49a0-98e8-0b763ca7f365)

### Expected Behavior

Transactions should be validated and executed at L1

### Current Behavior

Transactions are stuck at validation step.

### Additional Details

Logs: 5 errors from the server

`Oct 10 11:12:34 validium-testnet start.sh[2848344]: 2024-10-10T11:12:34.619335Z  INFO zksync_node_api_server::web3::metrics: Observed error code 3 (origin: App) for method eth_call`

`Oct 10 11:12:19 validium-testnet start.sh[2848344]:                     account: 0x8880…8878 does not have enough funds for for transferring tx.value: 2592000149028397183530.`

`Oct 10 11:12:19 validium-testnet start.sh[2848344]: 2024-10-10T11:12:19.742503Z  INFO zksync_node_api_server::tx_sender::gas_estimation: fee estimation failed on validation step.`

![Screenshot from 2024-10-10 16-28-08](https://github.com/user-attachments/assets/d98e3a8a-b20b-4e73-bf95-9fdbd3726c79)

![Screenshot from 2024-10-10 16-28-38](https://github.com/user-attachments/assets/5948ac45-85d1-4a3b-8fa1-2207e5435d5c)

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
It seems you are encountering issues with transactions getting stuck at the validation step and facing difficulties with L2 to L1 transactions while using a Validium based Zk hyperchain. This could be related to several factors including configuration settings, network issues, or problems with the custom base token setup.

Here are a few steps you can take to troubleshoot and potentially resolve the issue:

1. **Check Server and Prover Logs**: Ensure that both the server (zksync-era) and the prover (era-boojum) are running without errors. Look for any warnings or errors in the logs that might indicate what is causing the transactions to get stuck.

2. **Verify Network Configuration**: Since you are using holesky as L1, ensure that the network settings are correctly configured and that the ZkSync node can communicate effectively with the L1 network.

3. **Review Smart Contract Deployment**: Double-check the smart contracts deployed as part of your setup. Ensure they are correctly configured to handle the custom base token and that there are no issues in their logic that could be causing transactions to stall.

4. **Test with Simplified Setup**: If possible, try simplifying your setup or using a more standard configuration to determine if the issue is with the custom components or the basic infrastructure.

5. **Consult Documentation and Community**: Review the [ZKsync developer documentation](https://code.zksync.io/tutorials/custom-zk-chain/customizing-your-chain) again to ensure all steps were followed correctly. You might also want to reach out to the community or check if others have faced similar issues.

If these steps do not resolve the issue, you may need to provide more detailed logs or error messages for further assistance.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by dutterbutter** [comment]
@sachanayush47 can you ensure that the wallets in `wallet.yaml` have sufficient balances? It seems from the logs shared that `account: 0x8880…8878 does not have enough funds for for transferring tx.value: 2592000149028397183530.` Can you also clarify the usage of a `mock gpu prover`?

**Reply by sachanayush47** [reply]
> @sachanayush47 can you ensure that the wallets in `wallet.yaml` have sufficient balances? It seems from the logs shared that `account: 0x8880…8878 does not have enough funds for for transferring tx.value: 2592000149028397183530.` Can you also clarify the usage of a `mock gpu prover`?

All the wallets are funded with atleast 5 holEth. 
This address 0x8880…8878 is ecosystem's governor address. This 2592000149028397183530 results to 2592 eth which is very high.
While creating ecosystem, zk_inception cli asks for prover method. It has 2 options GPU and No proofs. I choose GPU and by default GPU prover is a mock prover which generates mock proofs to save compute resources.

**Reply by dutterbutter** [reply, marked as answer]
Thanks for the details! 

Yes the amount is exceptionally high but it seems thats where the issue is. What was nominator / denominator base token price and token address? In previous posts if I recall you set it at a 20:1 ETH ratio. This could explain some issues perhaps. I would try keeping it 1:1 to start and see if that works as intended. Then adjust accordingly.

**Reply by sachanayush47** [reply]
> Thanks for the details! 
> 
> Yes the amount is exceptionally high but it seems thats where the issue is. What was nominator / denominator base token price and token address? In previous posts if I recall you set it at a 20:1 ETH ratio. This could explain some issues perhaps. I would try keeping it 1:1 to start and see if that works as intended. Then adjust accordingly.

Yeah, ratio is set to 20:1
Sure, I will try with 1:1 ratio

**Reply by sachanayush47** [reply]
> Thanks for the details!
> 
> Yes the amount is exceptionally high but it seems thats where the issue is. What was nominator / denominator base token price and token address? In previous posts if I recall you set it at a 20:1 ETH ratio. This could explain some issues perhaps. I would try keeping it 1:1 to start and see if that works as intended. Then adjust accordingly.

Thanks, this has fixed the problem.
